OSPSuite.Presentation.Importer/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -0,0 +1,188 @@
+﻿using OSPSuite.Core.Importer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+
+namespace OSPSuite.Presentation.Importer.Core.DataFormat
+{
+   public abstract class AbstractColumnsDataFormat : IDataFormat
+   {
+      public abstract string Name { get; }
+      public abstract string Description { get; }
+      public IList<DataFormatParameter> Parameters { get; protected set; }
+
+      public bool SetParameters(IUnformattedData rawData, IReadOnlyList<ColumnInfo> columnInfos)
+      {
+         if (notCompatible(rawData, columnInfos))
+            return false;
+
+         setParameters(rawData, columnInfos);
+         return true;
+      }
+
+      protected bool notCompatible(IUnformattedData data, IReadOnlyList<ColumnInfo> columnInfos)
+      {
+         return (data.GetHeaders()
+            .Select(data.GetColumnDescription)
+            .Count(header => header.Level == ColumnDescription.MeasurementLevel.Numeric)) < columnInfos.Count(ci => ci.IsMandatory);
+      }
+
+      private void setParameters(IUnformattedData data, IReadOnlyList<ColumnInfo> columnInfos)
+      {
+         var keys = data.GetHeaders().ToList();
+         Parameters = new List<DataFormatParameter>();
+
+         var missingKeys = new List<string>();
+
+         extractQualifiedHeadings(keys, missingKeys, columnInfos, data);
+         extractNonQualifiedHeadings(keys, missingKeys, data);
+         extractGeneralParameters(keys, data);
+      }
+
+      protected abstract Func<int, string> extractUnits(string description, IUnformattedData data, List<string> keys);
+
+      private void extractQualifiedHeadings(List<string> keys, List<string> missingKeys, IReadOnlyList<ColumnInfo> columnInfos, IUnformattedData data)
+      {
+         if (keys == null) throw new ArgumentNullException(nameof(keys));
+         foreach (var header in columnInfos.Select(ci => ci.DisplayName))
+         {
+            var headerKey = keys.FirstOrDefault(h => h.ToUpper().Contains(header.ToUpper()));
```

Comment:
Done

Reply:
we changed it back since it was making tests red. If we do it all in the beginning it is complicated to then select the normal (lowercase) keys, so we get everything uppercase.